art_projected_gui/src/gui/ui_core_ros.py

```diff
@@ -411,14 +417,24 @@ def object_selected(self,  id,  selected):
                 pass
 
             poly_points = []
+            square_points = []
 
             self.program_vis.set_object(obj.object_type)
             self.select_object_type(obj.object_type)
 
+            # pozicie itemov na scene (tie dva co tam su momentalne)
             for obj in self.get_scene_items_by_type(ObjectItem):
                 poly_points.append(obj.get_pos())
+                square_points.append(obj.get_pos())
+
+
+            # !vytvori polygon po dvojkliku na item! #
+            # self.add_polygon(translate("UICoreRos", "PICK POLYGON"),  poly_points,  polygon_changed=self.polygon_changed)
+
 
-            self.add_polygon(translate("UICoreRos", "PICK POLYGON"),  poly_points,  polygon_changed=self.polygon_changed)
+            # ! ja som pridal ! #
+            self.add_square(translate("UICoreRos", "PICK SQUARE GRID"),  square_points,  square_changed=self.square_changed)
```

Comment:
Aktuálně ten polygon slouží jen na vymezení oblasti, odkud se mají brát objekty jednoho typu (proto "pick polygon"). Robot (resp. art_brain) pak umí chycený objekt (art_brain vybere z vybraného polygonu jeden konkrétní objekt) zatím jen položit na zadané místo (place_pose). Grid tedy má být alternativou k místu pro položení. Zatím byste mohl místo nastavení místa pro položení (add_place v ui_core_ros.py) vždy nastavovat grid. Ideálně by pak měla být možnost toto zadat buď v programu robota (https://github.com/robofit/ar-table-common/blob/master/art_msgs/msg/ProgramItem.msg), nebo to nechat vybrat uživatele.